src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -474,11 +474,19 @@ impl<'a,T:Payload> Ref<'a,T> {
             let remaining_ast_crumbs = ast_crumbs;
             Some(NodeFoundByAstCrumbs{node, ast_crumbs: remaining_ast_crumbs })
         } else {
-            let mut children = self.node.children.iter();
-            // Please be advised, that the `ch.ast_crumhs` is not a field of Ref, but Child, and
+            // Please be advised, that the `ch.ast_crumbs` is not a field of Ref, but Child, and
             // therefore have different meaning!
-            let next = children.find_position(|ch| {
+            let next = self.node.children.iter().find_position(|ch| {
                 !ch.ast_crumbs.is_empty() && ast_crumbs.starts_with(&ch.ast_crumbs)
+            }).or_else(|| {
+                // We do the second round to cover case of "prefix-like" nodes with
+                // `InsertionPoint(ExpectedArgument(_))`. See also docs for
+                // `generate::generate_expected_argument`.
```

Comment:
What is a "second round"? I feel these docs here are very important but I also feel they are too short and not very understandable for someone not knowing this part of code really, really well.

Reply:
This is second round of finding node, as I thought can be deduced from the code. I'll extend the comment a little.